tests/manage/z_cluster/test_rook_ceph_osd_flapping.py

```diff
@@ -0,0 +1,102 @@
+import logging
+import pytest
+import time
+
+
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.ocs.cluster import ceph_health_check
+from ocs_ci.ocs.resources import pod
+from ocs_ci.framework.pytest_customization.marks import brown_squad
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier2,
+    bugzilla,
+    skipif_ocs_version,
+    skipif_external_mode,
+    ignore_leftovers,
+)
+
+log = logging.getLogger(__name__)
+
+
+@brown_squad
+@tier2
+@ignore_leftovers
+@bugzilla("2223959")
+@skipif_external_mode
+@skipif_ocs_version("<4.14")
+@pytest.mark.polarion_id("OCS-5404")
+class TestRookCephOsdFlapping(ManageTest):
+    """
+    Test Rook Ceph OSD Flapping
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        def finalizer():
+            self.osd_pod_obj.delete()
+            ceph_health_check(tries=40, delay=30)
+
+        request.addfinalizer(finalizer)
+
+    def test_rook_ceph_osd_flapping(self):
+        """
+        Test Process:
+
+        1.Get osd id
+        2.Mark an osd down manually 6 times
+        3.Verify osd is down with "csph -s" command ["1 osds down"]
+        4.Verify osd log contains "The OSD pod will sleep for 24 hours." string
+        5.Reset osd pod [oc delete pod]
+        6.Verify ceph status is OK
+        """
+        log.info("Get One OSD ID")
+        osd_pod_objs = pod.get_osd_pods()
+        self.osd_pod_obj = osd_pod_objs[0]
+        osd_pod_id = pod.get_osd_pod_id(self.osd_pod_obj)
+        ct_pod = pod.get_ceph_tools_pod()
+        log.info(f"Mark an osd {osd_pod_id} down manually")
+        for _ in range(6):
+            time.sleep(5)
```

Comment:
I am not sure if we need to have a sleep time here, in the BZ verification I see that we have to do in " quick succession ".
Please check with Vishaka if there is a need for sleep here